themes/default/content/blog/organizational-patterns-automation-team/index.md

```diff
@@ -0,0 +1,83 @@
+---
+Using Pulumi is more than just writing code and components. In addition to common software development practices, there are also a number of success patterns related to how your company or team builds and deploys Pulumi programs to successfully build, deploy, and manage your infrastructure and applications. In this continuation of a series, I will explore one of these patterns - a specialized automation team.
+
+<!--more-->
+I want to approach this pattern with a bit of caution - there are patterns and anti-patterns related to having a single central team focused on automation.
+
+The anti-pattern is to have an "automation team" within your organization who is tasked with creating all infrastructure as code for the entire organization. This is a common pattern, but it is generally rife with issues.
+
+Understanding how to build and deploy a service or application requires a certain amount of domain knowledge of that service. If you have a single set of people across your entire company who need to write Pulumi programs for each and every application or service, you are now expecting that small group to not only be experts in Pulumi, but also every bit of infrastructure and software applications (both third party and created internally) across your organization.
+
+A more successful pattern is to have the automation team act as enablers and providers for the other groups within the company. This is a team of subject matter experts in using Pulumi, and not only do they have the knowledge to build and deploy infrastructure, but they are familiar with various patterns and good practices for how to build and deploy infrastructure.
+
+There are different mechanisms for how this automation team can enable others. One way is to embed them temporarily within a product team to work alongside them to build their automation. Then, when the product team is done, the automation team member can go and help another squad.
+
+Depending upon the size and skills of the company and this group, they may also be responsible for creating training materials for the other groups. These are usually based upon existing Pulumi training materials, but customized for the specific needs and culture of the company.
+
+## Reusable components
+
+This central automation team also often provides reusable components that might be shared across multiple development teams. This helps the individual teams to not have to "reinvent the wheel" and to bring in these components for their own particular needs.
+
+As a simple example, a component could be created (and published as a library to be consumed by product teams) that defines a Kubernetes deployment.
+
+```javascript
+import * as deploy from "@mycorp/deploy";
+
+const kuard new deploy.Deployment("kuard", {
+    image: "gcr.io/kuard-project/kuard:latest",
+    replicas: 1,
+    ports: [{
+        name: "http",
+        port: 8080,
+        targetPort: 8080,
```

Comment:
I’d omit this too and call it out. The application team don’t need to duplicate values because the abstractions cater for them.  The application team also didn’t need to apply any metadata, health checks, and environment variables because it can be inferred through convention, increasing velocity across the organization. When values are provided by the application team, they override convention and they can still iterate and experiment, providing feedback to the automation team.

Reply:
good point; the ports part isn't needed.